### PR TITLE
Check header file name collision when adding new library

### PR DESCRIPTION
**Pull request by Gisonrg**
Implement the front-end code to check default header file name when adding new library. Also add a new API to check for library meta using header file name.
If the header name already exists, now will populate the meta informations in the add new library form automatically and add it as a new version for the library.

### CODE REVIEW

File: Symfony/src/Codebender/LibraryBundle/Resources/views/Api/page-logic.html.twig [hunk 1]
```diff
@@ -44,6 +44,46 @@
         $('#jstreeDiv').jstree(true).refresh(false, true);
     }
 
+    function updateLibraryMeta(name)
+    {
+        var route = "{{ path('codebender_library_get_library_meta', {'authorizationKey': authorizationKey, 'library': "LIBRARYNAME"}) }}";
+        $.post( route.replace("LIBRARYNAME", name) )
```
Comment (spiliot) [inline Symfony/src/Codebender/LibraryBundle/Resources/views/Api/page-logic.html.twig]: No need to have spaces after and before parentheses.

File: Symfony/src/Codebender/LibraryBundle/Resources/views/Api/newLibForm.html.twig [hunk 2]
```diff
@@ -105,6 +105,24 @@
...
+    <!-- A modal used when header name collision happen -->
+    <div id="myModal" class="modal fade" tabindex="-1" role="dialog">
+        <div class="modal-dialog">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <h4 class="modal-title">Header file name already exists</h4>
+                </div>
+                <div class="modal-body">
+                    <p>The header file name of the library already exists in our system.</p>
+                    <p>Are you wish to continue and add as a new version for the existing library?</p>
```
Comment (spiliot) [inline Symfony/src/Codebender/LibraryBundle/Resources/views/Api/newLibForm.html.twig]: It looks like two questions were combined in one here :wink:

**Comment by Gisonrg** [comment]
@spiliot Done, please review :smiley: 
`getExternalExampleCode`, `getExampleCodeFromDir`, and `getExampleFilesFromDir` are functions of previous work, so I don't think I should modify the inside logic, but I did update their code style to fit PSR-2.

File: Symfony/src/Codebender/LibraryBundle/Resources/views/Api/newLibForm.html.twig (hunk 2 above)
Comment (Gisonrg) [inline Symfony/src/Codebender/LibraryBundle/Resources/views/Api/newLibForm.html.twig]: Hi John, so the two questions should be:
1. Continue as adding new library?
2. Continue as adding new version?
right?
(the third choice is "Cancel" :stuck_out_tongue: )

The later one is what we currently have here. But for the first one we might need to present an input box for user to let them enter name for the new library header? Or shall we just stop there and let user to modify the header file on their own in the file system?

Comment (spiliot) [inline Symfony/src/Codebender/LibraryBundle/Resources/views/Api/newLibForm.html.twig]: Only number 2 (and cancel) is applicable to the action a user can take at this point. We don't need to over-engineer things.

What I meant was far more subtle! The text in the question is "Are you wish to continue" which is like "Are you sure you want to continue" and "Do you wish to continue" combined :smiley:

Comment (Gisonrg) [inline Symfony/src/Codebender/LibraryBundle/Resources/views/Api/newLibForm.html.twig]: Haha, I see.
Maybe changed it to "Do you wish to continue and add as a new version for the existing library?" would be clearer.

Comment (thngkaiyuan) [inline Symfony/src/Codebender/LibraryBundle/Resources/views/Api/newLibForm.html.twig]: Perhaps "Do you wish to continue adding a new version to the library <library name>?"

Comment (Gisonrg) [inline Symfony/src/Codebender/LibraryBundle/Resources/views/Api/newLibForm.html.twig]: @thngkaiyuan Good one :+1:

**Pull request merged**